assets/src/block-editor/plugins/amp-preview-button.js

```diff
@@ -105,22 +105,49 @@ function writeInterstitialMessage( targetDocument ) {
 /**
  * A 'Preview AMP' button, forked from the Core 'Preview' button: <PostPreviewButton>.
  *
- * Rendered into the DOM with renderPreviewButton() in helpers/index.js.
- * This also moves the (non-AMP) 'Preview' button to before this, if it's not already there.
+ * Inserted into the DOM immediately after the 'Preview' button.
  *
  * @see https://github.com/WordPress/gutenberg/blob/95e769df1f82f6b0ef587d81af65dd2f48cd1c38/packages/editor/src/components/post-preview-button/index.js#L95-L200
  */
-class AMPPreview extends Component {
+class AmpPreviewButton extends Component {
 	/**
 	 * Constructs the class.
 	 *
 	 * @param {*} args Constructor arguments.
 	 */
 	constructor( ...args ) {
 		super( ...args );
-		this.moveButton = this.moveButton.bind( this );
-		this.openPreviewWindow = this.openPreviewWindow.bind( this );
+
 		this.buttonRef = createRef();
+		this.openPreviewWindow = this.openPreviewWindow.bind( this );
+
+		this.root = document.createElement( 'div' );
+		this.root.id = 'amp-wrapper-post-preview';
+
+		this.postPreviewButton = document.querySelector( `.${ POST_PREVIEW_CLASS }` );
```

Comment:
I suggest using the parent node instead of the preview button, because the preview button could be removed or replaced from outside. 

```suggestion
		const postPreviewButton = document.querySelector( `.${ POST_PREVIEW_CLASS }` );
		
		this.previewButtonParentNode = postPreviewButton ? postPreviewButton.parentNode : null;
```

(Also updating the checks elsewhere in the component.)

Reply:
I'm not sure what would be the benefit here. With this approach, we would have to include finding where the 'Preview' button is in the parent element, and _then_ insert our preview button next to it, no? I'm imagining something like:

```js
const postPreviewButton = ...;// additional logic to find preview button.

// Insert the AMP preview button immediately after the post preview button.
this.previewButtonParentNode.insertBefore( this.root, postPreviewButton.nextSibling );
```

Reply:
I was thinking if there is another plugin that replaces the core preview button, like:

```JS
const postPreviewButton = document.querySelector( `.${ POST_PREVIEW_CLASS }`;
const myPreviewButtonContainer = document.createElement( 'div' );
postPreviewButton.parentNode.replaceChild( mySubmitButtonContainer, postPreviewButton );

ReactDOM.createPortal( <MySpecialPreviewButton />, myPreviewButtonContainer );
```
This probably isn't the best idea, but then again this PR does something that comes close, so seeing a plugin that takes it a step further by replacing or even removing the core preview button wouldn't surprise me.

Given this, it might make sense to select the parent node in the first place, then insert the AMP preview button after the preview button if the preview button exists. 

Ultimately I would suggest we entirely replace this current approach with one that doesn't manipulate the DOM at all. E.g., using the `PluginPostStatusInfo` slotfill we could do this (maybe a little prettier):

<img width="283" alt="Screen Shot 2020-09-30 at 8 46 02 AM" src="https://user-images.githubusercontent.com/9020968/94693935-ebbbd580-02f9-11eb-8fa8-b4cfdcde78e1.png">

The placement is not as ideal as the top bar, but using the slotfill would be much more reliable, as we can expect that API to be stable while the editor DOM might change on any Gutenberg release.

Not saying we need to do this now, but it might be worth thinking about 😄

Reply:
> Given this, it might make sense to select the parent node in the first place, then insert the AMP preview button after the preview button if the preview button exists.

The intention is to eventually render the preview button in the 'Preview' dropdown, which should be possible once https://github.com/WordPress/gutenberg/pull/25430https://github.com/WordPress/gutenberg/pull/25430 is merged. However, for prior versions of Gutenberg that would not have this feature, we'd fallback to rendering the AMP preview button next to the 'Preview' button.

Reply:
Whoa, I didn't know that new slotfill was happening. Good to know! In that case, assuming we'll be following up, I'll mark this conversation as resolved.